fuzzing/fuzzer.go

```diff
@@ -348,16 +378,29 @@ func chainSetupFromCompilations(fuzzer *Fuzzer, testChain *chain.TestChain) (err
 		}
 	}
 
-	// Loop for all contracts to deploy
+	// Concatenate the target contracts and the predeployed contracts
+	contractsToDeploy := make([]string, 0)
+	contractsToDeploy = append(contractsToDeploy, fuzzer.config.Fuzzing.TargetContracts...)
+	for contractName, _ := range fuzzer.config.Fuzzing.PredeployedContracts {
+		contractsToDeploy = append(contractsToDeploy, contractName)
+	}
+
 	deployedContractAddr := make(map[string]common.Address)
-	for i, contractName := range fuzzer.config.Fuzzing.TargetContracts {
+	// Loop for all contracts to deploy
+	for i, contractName := range contractsToDeploy {
 		// Look for a contract in our compiled contract definitions that matches this one
 		found := false
 		for _, contract := range fuzzer.contractDefinitions {
 			// If we found a contract definition that matches this definition by name, try to deploy it
 			if contract.Name() == contractName {
+				// Concatenate constructor arguments, if necessary
 				args := make([]any, 0)
 				if len(contract.CompiledContract().Abi.Constructor.Inputs) > 0 {
+					// If the contract is a predeployed contract, throw an error because they do not accept constructor
+					// args.
+					if _, ok := fuzzer.config.Fuzzing.PredeployedContracts[contractName]; ok {
+						return fmt.Errorf("predeployed contracts cannot accept constructor arguments"), nil
+					}
```

Comment:
Technically, we could allow for predeployed contracts to take constructor args. However, currently echidna does not support it so for parity i thought of doing the same. It's easy to allow it though.